test/testuninitvar.cpp

```diff
@@ -4074,6 +4074,17 @@ class TestUninitVar : public TestFixture {
                         "    b(&s1);\n"
                         "}\n");
         ASSERT_EQUALS("", errout.str());
+
+        // # 9290
+        valueFlowUninit("struct A {\n"
+                        "    double x;\n"
+                        "};\n"
+                        "double b() {\n"
+                        "    A c;\n"
+                        "    c->x = 42;\n"
```

Comment:
That should be `c.x;`!?

Reply:
Oops, thats a mistake. If its a pointer, there should be an error. Let me add some more test cases.